ui/components/app/snaps/snap-ui-asset-selector/snap-ui-asset-selector.tsx

```diff
@@ -0,0 +1,152 @@
+import React, { FunctionComponent } from 'react';
+import { CaipAccountId, CaipChainId } from '@metamask/utils';
+
+import { SnapUISelector } from '../snap-ui-selector';
+
+import {
+  Display,
+  FlexDirection,
+  TextAlign,
+  TextColor,
+  TextVariant,
+  AlignItems,
+  BackgroundColor,
+  BlockSize,
+} from '../../../../helpers/constants/design-system';
+import {
+  Box,
+  Text,
+  AvatarToken,
+  BadgeWrapper,
+  AvatarNetwork,
+  AvatarNetworkSize,
+} from '../../../component-library';
+
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import { SnapUIAsset, useSnapAssetSelectorData } from './useSnapAssetDisplay';
+
+/**
+ * An option for the SnapUIAssetSelector.
+ *
+ * @param props - The component props.
+ * @param props.icon - The asset icon.
+ * @param props.symbol - The asset symbol.
+ * @param props.name - The asset name.
+ * @param props.balance - The asset balance.
+ * @param props.fiat - The asset balance in fiat.
+ * @param props.networkName - The network name.
+ * @param props.networkIcon - The network icon.
+ * @returns The Asset Selector option.
+ */
+const SnapUIAssetSelectorOption: FunctionComponent<SnapUIAsset> = ({
+  icon,
+  symbol,
+  name,
+  balance,
+  fiat,
+  networkName,
+  networkIcon,
+}) => (
+  <Box
+    display={Display.Flex}
+    alignItems={AlignItems.center}
+    width={BlockSize.Full}
+    gap={4}
+    style={{ overflow: 'hidden' }}
+  >
+    <Box alignItems={AlignItems.center}>
```

Comment:
It doesn't seem like this is properly aligned on my end:
![image](https://github.com/user-attachments/assets/e9b29aae-f717-4a57-938c-d27496ba9078)

Also the balance part is not being properly hidden 🤔